src/main/java/com/google/sps/data/UserInfo.java

```diff
@@ -32,7 +34,7 @@
   private final List<Long> created_events;
   private final List<Long> bookmarked_events;
   private Long current_challenge_id;
-  private final List<Integer> challenge_statuses;
+  private List<Integer> challenge_statuses;
```

Comment:
You switch between List<Integer> and ArrayList<Integer>. Stick with `ArrayList<Integer>` throughout the code, only dropping to the more abstract `List<Integer>` when truly needed.